internal/engine/generate/plpgsql.go

```diff
@@ -134,7 +134,12 @@ func (s *sqlGenerator) VisitExpressionVariable(p0 *parse.ExpressionVariable) any
 		// Postgres uses $1, $2, etc. for numbered parameters.
 
 		s.orderedParams = append(s.orderedParams, str)
-		return "$" + fmt.Sprint(len(s.orderedParams))
+
+		res := strings.Builder{}
+		res.WriteString("$")
+		res.WriteString(fmt.Sprint(len(s.orderedParams)))
```

Comment:
`fmt.Sprint` is slow and kinda overkill here.  I'd use `res.WriteString(strconv.Itoa(len(s.orderedParams)))` to skip all the printer's reflection that would end up doing that eventually.